life50x50/Form1.cs

```diff
@@ -0,0 +1,206 @@
+            menuStrip1.Items[1].Visible = true;
+            menuStrip1.Items[2].Enabled = false;
+            timer.Start();
+        }
+
+        private void TimerEventProcessor(Object o, EventArgs e)
+        {
+            timer.Stop();
+            gameMove();
+
+            stage++;
+            toolStripStatusLabel1.Text = "Поколение " + stage.ToString();
+            panel1.Refresh();
+
+            if (changes == 0 | totalAlive == 0)
+            {
+                MessageBox.Show("Игра окончена");
+                gamestarted = false;
+                menuStrip1.Items[0].Visible = true;
+                menuStrip1.Items[1].Visible = false;
+                menuStrip1.Items[2].Enabled = true;
+            }
+            else
+            {
+                timer.Enabled = true;
+            }            
+        }
+
+        private void gameMove()
+        {
+            changes = 0;
+            if (totalAlive > 0)
+            {
+                int total = 0;
+                bool[,] nextState = new bool[50, 50];
+
+                for (int i = 0; i < 50; i++)
+                {
+                    for (int j = 0; j < 50; j++)
+                    {
+                        int cur = checkNearest(i, j);
+                        if (gameboard[i, j] == false)
+                        {
+                            nextState[i, j] = false;
+                            if  (cur == 3)
+                            {
+                                nextState[i, j] = true;
+                                total++;
+                                changes++;
+                            }
+                        }
+                        else
+                        {
+                            nextState[i, j] = true;
+                            total++;
+                            if (cur < 2 ^ cur > 3)
+                            {
+                                nextState[i, j] = false;
+                                changes++;
+                                total--;
+                            }
+                        }
+
+                    }
+                }
+
+                gameboard = nextState;
+                totalAlive = total;
+            }            
+        }
+
+        private int checkNearest(int i, int j)
+        {
+            int total = 0;
+            int top = i == 0 ? 49 : i - 1;
+            int bot = i == 49 ? 0 : i + 1;
+            int left = j == 0 ? 49 : j - 1;
+            int right = j == 49 ? 0 : j + 1;
+
+            if (gameboard[top, left] == true) { total++; }
+            if (gameboard[top, j] == true) { total++; }
+            if (gameboard[top, right] == true) { total++; }
+            if (gameboard[i, left] == true) { total++; }
+            if (gameboard[i, right] == true) { total++; }
+            if (gameboard[bot, left] == true) { total++; }
+            if (gameboard[bot, j] == true) { total++; }
+            if (gameboard[bot, right] == true) { total++; }

```

Comment:
Можно было написать покороче
```suggestion
            if (gameboard[top, left]) { total++; }
            if (gameboard[top, j]) { total++; }
            if (gameboard[top, right]) { total++; }
            if (gameboard[i, left]) { total++; }
            if (gameboard[i, right]) { total++; }
            if (gameboard[bot, left]) == true) { total++; }
            if (gameboard[bot, j]) { total++; }
            if (gameboard[bot, right]) { total++; }

```